node-forge/index.d.ts

```diff
@@ -234,4 +236,39 @@ declare module "node-forge" {
             }
         }
     }
+
+    namespace pkcs12 {
+
+        interface BagsFilter {
+            localKeyId?: string;
+            localKeyIdHex?: string;
+            friendlyName?: string;
+            bagType?: string;
+        }
+
+        interface Bag {
+            type: string;
+            attributes: any;
+            key?: pki.Key;
+            cert?: pki.Certificate;
+            asn1: asn1.Asn1
+        }
+
+        interface Pkcs12Pfx {
+            version: string;
+            safeContents: [{
+                encrypted: boolean;
+                safeBags: Bag[];
+            }];
+            getBags: (filter: BagsFilter) => {
+                [key: string]: Bag[];
+                localKeyId?: Bag[];
+                friendlyName?: Bag[];
+            };
+            getBagsByFriendlyName: (fiendlyName: string, bagType: string) => Bag[]
+            getBagsByLocalKeyId: (localKeyId: string, bagType: string) => Bag[]
+        }
+
+        function pkcs12FromAsn1(obj:any, strictOrPassword: boolean|string, password?: string) : Pkcs12Pfx;
```

Comment:
It might be best to use function overloading here instead of providing the single function to handle multiple situations.